### PR TITLE
changed: move application settings handling to separate class

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Move application settings handling to separate class

## Motivation and context
Reduce complexity of Application.cpp

## How has this been tested?
It builds and runs

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/Application.h [hunk 1]
```diff
@@ -229,14 +222,6 @@ friend class CAppInboundProtocol;
 
 protected:
   bool OnSettingsSaving() const override;
```
Comment (ksooo) [inline xbmc/Application.h:224]: Could you explain why this one was not factored out to `CApplicationSettingsHandling`?

**Review by ksooo — COMMENTED** [review]
Nice improvement.

File: xbmc/Application.h (hunk 1 above)
Comment (ksooo) [reply xbmc/Application.h:224]: `CApplication` is no longer directly derived from `ISettingsHandler`, but inherits this interface from `CApplicationSettingsHandling`. This is why this overrode still works. Is this by intention?

**Review by garbear — APPROVED** [review]

File: xbmc/Application.h (hunk 1 above)
Comment (ksooo) [reply xbmc/Application.h:224]: I see.

**Review by ksooo — APPROVED** [review]
👍

**Pull request merged**